src/LegacyPaymentGateways/Adapters/LegacyDonationData.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Give\LegacyPaymentGateways\Adapters;
+
+use Give\PaymentGateways\DataTransferObjects\GatewayPaymentData;
+
+/**
+ * @unreleased
+ */
+class LegacyDonationData
```

Comment:
@kjohnson Maybe we do not require this. I find out that you are using this for function `give_stripe_prepare_metadata`. Can we use this opportunity to deprecate and create a new one that will use `GatewayPaymentData`?
If you do not want to refactor code then for this reason I added `legacyDonationData` in `FormData` which is only for internal use. I am coping the same property to create `GatewayPaymentData` which I am using to gracefully deprecate filter and action hook.

Let me know your thoughts.

Reply:
I think this is outside of the current scope. The goal here is to update to use the new Gateway API. I'm less concerned with the internals of the components used by the gateway - at this time - except where it prevents testing the new code.